File: tools/monty-checks/bench/worker.ts
import {DurableObject} from 'cloudflare:workers';

export class Counter extends DurableObject {
  ping() { return this.ctx.blockConcurrencyWhile(() => 1); }
  increment() {
    return this.ctx.blockConcurrencyWhile(() => this.ctx.storage.transactionSync(() => {
      const value = (this.ctx.storage.kv.get('count') ?? 0) + 1;
      this.ctx.storage.kv.put('count', value);
      return value;
    }));
  }
}

export default {async fetch(request, env) {
  const path = new URL(request.url).pathname;
  if (!['/hello','/echo','/increment','/pause','/object','/fetch'].includes(path)) return new Response('unknown handler',{status:404});
  if (request.method !== 'POST') return new Response('POST required',{status:405});
  const args = await request.json();
  switch (path) {
    case '/hello': return new Response(`Hello, ${args.name ?? 'world'}!`, {headers:{'content-type':'text/plain; charset=utf-8'}});
    case '/echo': return Response.json({value:args.value});
    case '/increment': return Response.json({value:await env.COUNTER.getByName(args.id).increment()});
    case '/object': return Response.json({value:await env.COUNTER.getByName(args.id).ping()});
    case '/fetch': return await fetch(args.url, {method:'POST', body:JSON.stringify({value:'x'.repeat(16384)})});
    case '/pause':
      await new Promise(resolve => setTimeout(resolve,10));
      return new Response('done',{headers:{'content-type':'text/plain; charset=utf-8'}});
  }
}};

File: tools/monty-checks/bench/worker.py
from celld import Context, Response


def hello(name: str = "world") -> str:
    return f"Hello, {name}!"


def echo(value: str) -> dict[str, str]:
    return {"value": value}


def increment(ctx: Context, id: str) -> dict[str, int]:
    return {"value": Counter(id, ctx).increment()}


async def pause(ctx: Context) -> str:
    await ctx.sleep(0.01)
    return "done"


class Counter:
    def __init__(self, id: str, ctx: Context):
        self.id = id
        self._ctx = ctx

    def ping(self) -> int:
        return 1

    def increment(self) -> int:
        with self._ctx.storage.transaction() as storage:
            previous = storage.get("count", 0)
            assert isinstance(previous, int)
            value = previous + 1
            storage.set("count", value)
        return value


def object(ctx: Context, id: str) -> dict[str, int]:
    return {"value": Counter(id, ctx).ping()}


async def fetch(ctx: Context, url: str) -> Response:
    return await ctx.fetch(url, method="POST", body='{"value":"' + "x" * 16384 + '"}')
